File: Detect Capital.java
class Solution {
    public boolean allUpperCase(String word) {
        for ( int i = 0 ; i < word.length() ; i++ ) {
            if ( !(word.charAt(i) >= 'A' && word.charAt(i) <= 'Z') ) {
                return false ;
            }
        }
        return true ;
    }

    public boolean allLowerCase(String word) {
        for ( int i = 0 ; i < word.length() ; i++ ) {
            if ( !(word.charAt(i) >= 'a' && word.charAt(i) <= 'z') ) {
                return false ;
            }
        }
        return true ;
    }

    public boolean firstUpperCase(String word) {
        for ( int i = 0 ; i < word.length() ; i++ ) {
            if ( i == 0 && !(word.charAt(i) >= 'A' && word.charAt(i) <= 'Z') ) {
                return false ;
            }
            if ( i != 0 && !(word.charAt(i) >= 'a' && word.charAt(i) <= 'z') ) {
                return false ;
            }
        }
        return true ;
    }

    public boolean detectCapitalUse(String word) {
        return (allUpperCase(word) == true || allLowerCase(word) == true || firstUpperCase(word) == true) ;
    }
}

File: Detect Capital.cpp
/*

Space Complexity : O(1)
Time Complexity : O(n)

*/

class Solution {
public:
    bool allUpperCase(string word) {
        for ( int i = 0 ; i < word.length() ; i++ ) {
            if ( !(word[i] >= 'A' && word[i] <= 'Z') ) {
                return false ;
            }
        }
        return true ;
    }

    bool allLowerCase(string word) {
        for ( int i = 0 ; i < word.length() ; i++ ) {
            if ( !(word[i] >= 'a' && word[i] <= 'z') ) {
                return false ;
            }
        }
        return true ;
    }

    bool firstUpperCase(string word) {
        for ( int i = 0 ; i < word.length() ; i++ ) {
            if ( i == 0 && !(word[i] >= 'A' && word[i] <= 'Z') ) {
                return false ;
            }
            if ( i != 0 && !(word[i] >= 'a' && word[i] <= 'z') ) {
                return false ;
            }
        }
        return true ;
    }

    bool detectCapitalUse(string word) {
        return (allUpperCase(word) == true || allLowerCase(word) == true || firstUpperCase(word) == true) ;
    }
};